test-tools/aot-analyzer/src/aot-reader.cc

```diff
@@ -0,0 +1,233 @@
+#include "aot-reader.h"
+
+#include <cstring>
+
+#include "analyzer-error.h"
+#include "common.h"
+
+namespace analyzer {
+
+AoTReader::AoTReader(const char* file_name, ObjdumpOptions options) : BinaryReader(file_name, options)
+{
+    memset(&target_info_, 0, sizeof(AOTTargetInfo));
+}
+
+Result AoTReader::CheckMagic() {
+  uint32_t magic = 0;
+  CHECK_RESULT(ReadU32(&magic, "magic"));
+  ERROR_UNLESS(magic == AOT_MAGIC_NUMBER, "bad magic value");
+
+  uint32_t version = 0;
+  CHECK_RESULT(ReadU32(&version, "version"));
+  ERROR_UNLESS(version == AOT_CURRENT_VERSION,
+               "bad AoT file version: %#x (expected %#x)", version,
+               AOT_CURRENT_VERSION);
+  return Result::Ok;
+}
+
+Result AoTReader::Scan() {
+  CHECK_RESULT(CheckMagic());
+  CHECK_RESULT(ParseTargetInfoSection());
+  aot_get_module_mem_consumption((AOTModule *)module_, &mem_conspn_);
+  return Result::Ok;
+}
+
+Result AoTReader::ParseTargetInfoSection() {
+  uint32_t section_id = 0;
+  CHECK_RESULT(ReadU32(&section_id, "section_id"));
+  ERROR_UNLESS(section_id == AOT_SECTION_TYPE_TARGET_INFO, "bad section id");
+  uint32_t section_size = 0;
+  CHECK_RESULT(ReadU32(&section_size, "section_size"));
```

Comment:
Since we already load informations from `aot_module`, do we still need to read the raw bytes?

Reply:
It is indeed somewhat repetitive to read raw bytes again here.
The `AoTModule` structure does not contain the `targetInfo`, and I cannot obtain the `targetInfo` information of the AoT file using the existing warmr interface.